File: mathenger-api/src/main/java/com/example/mathengerapi/models/Message.java
package com.example.mathengerapi.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "messages")
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Message {
    @Id
    @GeneratedValue
    private Long id;
    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false)
    private Account author;
    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private Account sender;
    @CreationTimestamp
    private LocalDateTime time;
    @Lob
    private String text;
}

File: mathenger/Mathenger/Mathenger/Models/Message.cs
using System;
using Newtonsoft.Json;

namespace Mathenger.models
{
    public class Message : BaseViewModel
    {
        [JsonProperty("id")] public long Id { get; set; }
        [JsonProperty("author")] public Account Author { get; set; }
        [JsonProperty("sender")] public Account Sender { get; set; }
        [JsonProperty("text")] public string Text { get; set; }
        [JsonProperty("time")] public DateTime Time { get; set; }
    }
}